caffe2/operators/onnxifi_op.cc

```diff
@@ -151,42 +151,63 @@ bool OnnxifiOp<float, CPUContext>::RunOnDevice() {
     SetOutputTensorDescriptorTypeAndBuffer(
         type, output_tensor, &tensor_descriptor);
   }
-
-  CAFFE_ENFORCE_EQ(
-      lib_->onnxSetGraphIO(
-          graph_,
-          input_desc_.size(),
-          input_desc_.data(),
-          output_desc_.size(),
-          output_desc_.data()),
-      ONNXIFI_STATUS_SUCCESS);
-
-  onnxMemoryFenceV1 input_fence;
-  input_fence.tag = ONNXIFI_TAG_MEMORY_FENCE_V1;
-  input_fence.type = ONNXIFI_SYNCHRONIZATION_EVENT;
-  CAFFE_ENFORCE_EQ(
-      lib_->onnxInitEvent(backend_, &input_fence.event),
-      ONNXIFI_STATUS_SUCCESS);
-  onnxMemoryFenceV1 output_fence;
-  output_fence.tag = ONNXIFI_TAG_MEMORY_FENCE_V1;
-  output_fence.type = ONNXIFI_SYNCHRONIZATION_EVENT;
-
-  // Call the asycn run on backend, singal event on input fence and wait for the
-  // event on output fence
-  CAFFE_ENFORCE_EQ(
-      lib_->onnxSignalEvent(input_fence.event), ONNXIFI_STATUS_SUCCESS);
-  CAFFE_ENFORCE_EQ(
-      lib_->onnxRunGraph(graph_, &input_fence, &output_fence),
-      ONNXIFI_STATUS_SUCCESS);
-  CAFFE_ENFORCE_EQ(
-      lib_->onnxWaitEvent(output_fence.event), ONNXIFI_STATUS_SUCCESS);
-
-  // Destroy the event objects
-  CAFFE_ENFORCE_EQ(
-      lib_->onnxReleaseEvent(input_fence.event), ONNXIFI_STATUS_SUCCESS);
-  CAFFE_ENFORCE_EQ(
-      lib_->onnxReleaseEvent(output_fence.event), ONNXIFI_STATUS_SUCCESS);
-
+  bool ext_success = false;
+#ifdef ONNXIFI_ENABLE_EXT
```

Comment:
onnxGetBackendInfo ONNXIFI method has a property for `ONNXIFI_BACKEND_EXTENSIONS` where is it used?

Reply:
I am not quite get it... Could you please tell me more about this in detail ? Thnaks!

Reply:
As part of ONNXIFI interface and `onnxGetBackendInfo` call, ONNXIFI implementation can provide which extensions are supported by implementation and returns it in `ONNXIFI_BACKEND_EXTENSIONS`.

It would be natural for caller to check if particular extension is supported via `onnxGetBackendInfo` call.

Reply:
Oh! I think you are right!
Then, let's say, ONNXIFI_EXT_ENABLE only indicate if this onnx is supporting extension or not, and onnxGetBackendInfo indicate if our backend is supporting extension. That's pretty cool.